dxr/plugins/clang/dxr-index.cpp

```diff
@@ -1087,7 +1095,7 @@ class IndexConsumer : public ASTConsumer,
             text[i] != '\t' && text[i] != '\n')
           text[i] = '?';
       }
-      recordValue("text", text, true);
+      macromap.insert(make_pair(nameStart, text));
```

Comment:
> It seems like this will keep the full text of any macro body in memory forever.

Isn't there one IndexConsumer per translation unit compilation, and that IndexConsumer gets freed as soon as that compilation completes?  That's what I've been thinking at least.

I did an m-c compile with some logging just to make sure - I never see the logging in the destructor for IndexConsumer that I added appear (I assume it's just getting quelched or appears too late - I do see IndexConsumer creation for each compilation unit), but it gave some other data that might be helpful here:
of 3028 translation units that had a macro def, the average final length of macromap was 1,692, the median was 1,440, and the max was 13,051, i.e. in m-c there are at most 13,051 macro defs in any given translation unit.  Those numbers are bigger than I was expecting, but still, if you take an average macro-def string of 800 bytes (10 lines average? I would guess that's high?), say, then with 13,051 entries, the largest macromap would be roughly 13,051 * (4 byte key (SourceLocation) + 800 byte macro-def-string + 10(?) bytes overhead) = 10,623,514 bytes or about 10.5M extra memory use, and the average compile would use about 1.4M extra memory for macromap.  And then you might have a bunch of compiles running at once, so maybe, I don't know, an extra 10-50 Mb?

Also on my machine with 8G ram I didn't have any problems compiling m-c under dxr (no swapping that I ever noticed, not even close actually), and my running firefox actually always had more memory use than any of the compiles going on as far as I saw, so I would be inclined to say it seems okay to me (if those numbers are anywhere close...).  But if there's more room to spare in the indexing step (is there? the impression I'd had in the past was that parts of the indexing step were using more memory than the compilation step), I could move it there.